### PR TITLE
chore: add release scripts

### PR DESCRIPTION
**Pull request by jye-sf**
One step closer to a 1-button release process.
The remaining work involves git automation around branch protection, pushing the actual changes to the remote, and changelog generation.

### CODE REVIEW

**Comment by jye-sf** [comment]
The current general release process for our repos will be:

```
// Version update
yarn release:version

// github related tasks
git push origin <branch>
git push origin <tag>

// Publish
yarn release:publish
```

Github related tasks require unlocking and re-locking branch protections. Will look into having CircleCI handle that.

**Review by nolanlawson — APPROVED** [review]

**Comment by jye-sf** [comment]
https://app.circleci.com/pipelines/github/salesforce/eslint-plugin-lwc/281/workflows/a3b90f6a-e1a7-4823-9313-7d8f5e5afa24/jobs/384 for sample run of the deploy job.

**Comment by jye-sf** [comment]
With the CircleCI script, our general release process should now be:

```
// Version update
yarn release:version

// github related tasks
git push origin <branch>
git push origin <tag>

// Manually create release with release notes
```

CircleCI should now handle the publish.

**Comment by jye-sf** [comment]
Work to automate changelog generation and release notes will be done in a separate PR.

File: package.json [hunk 1]
```diff
@@ -15,7 +15,9 @@
     "format": "prettier --write --ignore-path .gitignore \"**/*.{js,md}\"",
     "format:check": "prettier --check --ignore-path .gitignore \"**/*.{js,md}\"",
     "lint": "eslint lib/ test/",
-    "test": "nyc mocha"
+    "test": "nyc mocha",
+    "release:version": "npm version",
+    "release:publish:ci": "npm publish --registry=https://registry.npmjs.org --dry-run"
```
Comment (jye-sf) [inline package.json:20]: Will remove the `--dry-run` tag AFTER review for safety.

**Review by nolanlawson — APPROVED** [review]

**Pull request merged**